lib/icingadb/icingadb-objects.cpp

```diff
@@ -1722,15 +1722,23 @@ void IcingaDB::SendStartedDowntime(const Downtime::Ptr& downtime)
 	}
 
 	if (downtime->GetFixed()) {
+		auto startTime = downtime->GetStartTime();
+		auto endTime = downtime->GetEndTime();
+		auto duration = endTime - startTime;
+
 		xAdd.emplace_back("start_time");
-		xAdd.emplace_back(Convert::ToString(TimestampToMilliseconds(downtime->GetStartTime())));
+		xAdd.emplace_back(Convert::ToString(TimestampToMilliseconds(startTime)));
 		xAdd.emplace_back("end_time");
-		xAdd.emplace_back(Convert::ToString(TimestampToMilliseconds(downtime->GetEndTime())));
+		xAdd.emplace_back(Convert::ToString(TimestampToMilliseconds(endTime)));
+		xAdd.emplace_back("duration");
+		xAdd.emplace_back(Convert::ToString(duration));
 	} else {
 		xAdd.emplace_back("start_time");
 		xAdd.emplace_back(Convert::ToString(TimestampToMilliseconds(downtime->GetTriggerTime())));
 		xAdd.emplace_back("end_time");
 		xAdd.emplace_back(Convert::ToString(TimestampToMilliseconds(downtime->GetTriggerTime() + downtime->GetDuration())));
+		xAdd.emplace_back("duration");
+		xAdd.emplace_back(Convert::ToString(downtime->GetDuration()));
```

Comment:
Do we want the duration to be in seconds or milliseconds?

Reply:
Guess what, it's inconsistent already :(

https://github.com/Icinga/icinga2/blob/6b3a2ee42993f34226ef53db0d5773746c924bc5/lib/icingadb/icingadb-objects.cpp#L1333
https://github.com/Icinga/icinga2/blob/6b3a2ee42993f34226ef53db0d5773746c924bc5/lib/icingadb/icingadb-objects.cpp#L1700
https://github.com/Icinga/icinga2/blob/6b3a2ee42993f34226ef53db0d5773746c924bc5/lib/icingadb/icingadb-objects.cpp#L1786

Reply:
> Do we want the duration to be in seconds or milliseconds?

In DB Web it is expected to be seconds if I am not wrong. cc @nilmerg

Reply:
If it was computed as `downtime.scheduled_end_time - downtime.scheduled_start_time` before, it got milliseconds.

Reply:
Oh yes, because these two columns are already in milliseconds. I just looked in the DB web code and the function which formats the downtime duration expects the value to be in seconds.

Reply:
Web can easily change this. I often wondered why some values are seconds and others are milliseconds.